libs/pubsub/query/query.go

```diff
@@ -195,6 +197,28 @@ func (q *Query) Matches(events map[string][]string) bool {
 			op = OpEqual
 		case rulecontains:
 			op = OpContains
+		case ruleexists:
+			op = OpExists
+			if strings.Contains(tag, ".") {
+				// Searching for a full "type.attribute" event.
+				_, ok := events[tag]
+				if !ok {
+					return false
+				}
+			} else {
+				foundEvent := false
+
+			loop:
+				for k := range events {
+					if strings.Index(k, tag) == 0 {
```

Comment:
why Index? Shouldn't we use strict comparison here ?

```suggestion
					if k == tag {
```

Reply:
This is the case where the query is the "event type", e.g. just "slash". The contents of the `events` slice are the fully qualified event attributes, e.g. `slash.reason`. The index is used to verify that the event starts with the tag from the query.

Reply:
Won't it lead to false positives? I.e. `sl EXISTS` will work for `slash.reason`, but I doubt that it's what we want.

Reply:
`sl EXISTS` is valid syntax in the current implementation. Not sure whether I would consider it a problem.

Reply:
This needs to be reflected both in the docs https://github.com/tendermint/tendermint/blob/2df4ca0f3924961dfbd3cd66e1b530fd20909640/rpc/swagger/swagger.yaml#L155 and in the tests. Thank you 🙇